plugins/org.eclipse.n4js.ide/src/org/eclipse/n4js/ide/xtext/server/build/XWorkspaceBuilder.java

```diff
@@ -435,7 +435,9 @@ private void mergeWithUnreportedDeltas(List<IResourceDescription.Delta> newDelta
 					toBeConsideredDeltas.remove(unreportedDelta);
 					IResourceDescription _old = unreportedDelta.getOld();
 					IResourceDescription _new = newDelta.getNew();
-					toBeConsideredDeltas.add(new DefaultResourceDescriptionDelta(_old, _new));
+					if (_old != null || _new != null) {
+						toBeConsideredDeltas.add(new DefaultResourceDescriptionDelta(_old, _new));
+					}
```

Comment:
as discussed: consider keeping the newDelta if `_old == _new == null` to have dependent resources to be built